odoo/cli/code_upgrade.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+import argparse
+
+import os.path
+from importlib.machinery import SourceFileLoader
+from pathlib import Path
+from typing import Iterator
+from types import ModuleType
+
+from . import Command
+
+"""
+Script method is add into the /upgrade/code directory.
+Every script should be in a file corresponding to his upgrade version (script for the )
+
+"""
+
+AVAILABLE_EXT = ('py', 'js', 'css', 'scss', 'xml', 'csv')
+
+
+class FileDescriptor:
```

Comment:
The purpose of this class is to provide some abstraction for getting and setting a file's content, possibly without writing it for real.  So it's not a file _descriptor_, which refers to a specific OS object.  I suggest to call this a `FileAccessor`.